java/vector/src/test/java/org/apache/arrow/vector/file/TestArrowFile.java

```diff
@@ -156,11 +192,36 @@ public void testWriteReadComplex() throws IOException {
         }
       }
     }
+
+    // Read from stream.
+    try (
+        BufferAllocator readerAllocator = allocator.newChildAllocator("reader", 0, Integer.MAX_VALUE);
+        ByteArrayInputStream input = new ByteArrayInputStream(stream.toByteArray());
+        ArrowStreamReader arrowReader = new ArrowStreamReader(input, readerAllocator);
+        BufferAllocator vectorAllocator = allocator.newChildAllocator("final vectors", 0, Integer.MAX_VALUE);
+        MapVector parent = new MapVector("parent", vectorAllocator, null)
+        ) {
+      arrowReader.init();
+      Schema schema = arrowReader.getSchema();
+      LOGGER.debug("reading schema: " + schema);
+
+      try (VectorSchemaRoot root = new VectorSchemaRoot(schema, vectorAllocator)) {
+        VectorLoader vectorLoader = new VectorLoader(root);
+        while (true) {
+          try (ArrowRecordBatch recordBatch = arrowReader.nextRecordBatch()) {
```

Comment:
Yea, we should either have the file and stream reader both implement the same interface (ArrowIpcReader?) or have a higher level object that does the same thing.

These tests are not quite collapsable that way though as the existing file based tests do verify some of the internals like block alignment and what not.